src/web/Content/client/dumpling.py

```diff
@@ -880,16 +879,46 @@ def _download_dump(self, dir, dumpManifest):
 
         return dumplingDir
     
+    def HangCheck(self, config):
+        if config.upload:
+            self.HangAndUpload(config)
+        else:
+            self.Hang(config)
+
     def Hang(self, config):
-        if os.path.exists(config.dbgpath) and os.path.isdir(config.outpath):
-            CommandProcessor._create_hang_dump(config.pid, config.outpath, config.dbgpath)
+        if os.path.exists(config.dbgpath) and os.path.isdir(config.outpath):            
             process = psutil.Process(int(config.pid))
+            parent_dump_folder = os.path.join(config.outpath, "process" + str(config.pid))
+
+            if not os.path.exists(parent_dump_folder):
+                os.makedirs(parent_dump_folder)
+            else:
+                shutil.rmtree(parent_dump_folder)
+                os.makedirs(parent_dump_folder)
+
+            CommandProcessor._create_hang_dump(config.pid, parent_dump_folder, config.dbgpath)
             for child_process in process.children(recursive=True):
-                CommandProcessor._create_hang_dump(str(child_process.pid), config.outpath, config.dbgpath)
+                CommandProcessor._create_hang_dump(str(child_process.pid), config.outpath, parent_dump_folder)
         else:
             path = config.dbgpath if os.path.exists(config.outpath) else config.outpath
             Output.Critical('Invalid Path %s' % path)
-            
+
+    def HangAndUpload(self, config):
+        self.Hang(config)
+        parent_dump_folder = os.path.join(config.outpath, "process" + str(config.pid))
+        
+        if config.incpaths is None:
+            config.incpaths = [parent_dump_folder]
+        
+        if config.dumppath is None:
+            dump_file_name = "memdump" + str(config.pid) + ".dmp"
+            config.dumppath = os.path.join(parent_dump_folder, dump_file_name)
+        
+        if config.displayname is None:
+            config.displayname = "HungProcess" + str(config.pid);
```

Comment:
Hopefully we normally pass in a better name, that includes the test library name?